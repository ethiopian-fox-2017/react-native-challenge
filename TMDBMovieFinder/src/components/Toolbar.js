import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  TouchableHighlight
} from 'react-native'

import { styles } from '../styles'

export default class Toolbar extends React.Component {
  render() {
    const { navigation } = this.props
    return (
      <View style={styles.toolbar}>
        <Text style={styles.textTitle}>Movie Finder</Text>
        <TouchableHighlight
          onPress={() => navigation.navigate('About', {name: 'About'})}
          underlayColor="rgba(255,255,255,0.5)"
        >
          <Text style={styles.textTitle}>About</Text>
        </TouchableHighlight>
      </View>
    )
  }
}

Toolbar.propTypes = {
  navigation: PropTypes.object.isRequired
}