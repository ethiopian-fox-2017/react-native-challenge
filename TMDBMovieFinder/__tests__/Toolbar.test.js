import 'react-native'
import React from 'react'

import Toolbar from '../src/components/Toolbar'
import renderer from 'react-test-renderer'

it('Render Toolbar', () => {
  const wrapper = renderer.create(
    <Toolbar />
  ).toJSON()
  expect(wrapper).toMatchSnapshot()
})