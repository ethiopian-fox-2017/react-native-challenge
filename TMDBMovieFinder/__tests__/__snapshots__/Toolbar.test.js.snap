// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Toolbar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#01d277",
      "flexDirection": "row",
      "height": "10%",
      "justifyContent": "space-between",
      "paddingLeft": "5%",
      "paddingRight": "5%",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#073554",
        "fontSize": 15,
        "margin": 5,
      }
    }
  >
    Movie Finder
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#073554",
          "fontSize": 15,
          "margin": 5,
        }
      }
    >
      About
    </Text>
  </View>
</View>
`;
